### PR TITLE
Mv phi and fluid/test To test dir

### PR DESCRIPTION
**Pull request by tianshuo78520a**
### PR types

Others
### PR changes

Others
### Describe

修改Paddle单测目录，所有单测将放在/Paddle/test目录下
PR先进行phi，和一些Python单测迁移工作。

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -699,6 +699,7 @@ if(WITH_STRIP)
 endif()
 
 add_subdirectory(paddle)
+add_subdirectory(test)
```
Comment (zhiqiu) [inline CMakeLists.txt]: move it to the line after python, i.e, line 705

File: paddle/fluid/jit/layer_test.cc [hunk 2]
```diff
@@ -191,7 +191,7 @@ TEST(GpuLayerTest, Construct) {
 TEST(GpuLayerTest, Clone) {
   auto place = phi::GPUPlace();
 
-  std::string path = "./multi_program_load/export";
+  std::string path = "../multi_program_load/export";
```
Comment (zhiqiu) [inline paddle/fluid/jit/layer_test.cc]: what is the absolute path?

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,110 @@
+set(CC_TESTS_DIR
+    ${PADDLE_BINARY_DIR}/test/cpp
+    CACHE INTERNAL "c++ tests directory")
+set(PYTHON_TESTS_DIR
+    ${PADDLE_BINARY_DIR}/test
+    CACHE INTERNAL "python tests directory")
+
+add_subdirectory(cpp)
+add_subdirectory(old_test)
```
Comment (zhiqiu) [inline test/CMakeLists.txt]: old_test -> legacy_test?

File: paddle/CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 set(CC_TESTS_DIR
-    ${PADDLE_BINARY_DIR}/paddle/tests
+    ${PADDLE_BINARY_DIR}/test
```
Comment (zhiqiu) [inline paddle/CMakeLists.txt]: redefined in test/CMakeLists.txt

File: paddle/CMakeLists.txt [hunk 5]
```diff
@@ -48,93 +48,3 @@ set(POSTFIX ".so")
 if(WIN32)
   set(POSTFIX ".dll")
 endif()
```
Comment (zhiqiu) [inline paddle/CMakeLists.txt]: line 40-50 is no need here, and the comments should be moved to test/CMakeLists.txt

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**